### PR TITLE
Show correct default branch in repository config

### PR DESCRIPTION
**Pull request by eheimbuch**
## Proposed changes

Show correct default branch if no default branch is configured in repo config yet. Before the first branch (alphabetically) was shown.

### Your checklist for this pull request

**Contributor**:
- [x] PR is well described and the description can be used as a commit message on squash
- [x] Related issues linked to PR if existing and labels set
- [x] Target branch is not master (in most cases develop should bet the target of choice) 
- [x] Code does not conflict with target branch
- [ ] New code is covered with unit tests
- [ ] New ui components are tested inside the storybook (module ui-components only) 
- [x] [Changelog entry file](https://github.com/scm-manager/changelog#changelog-entry-files) created in `gradle/changelog` or CHANGELOG.md is updated for plugins
- [ ] Documentation updated (only necessary for new features or changed behaviour)

**Reviewer**:
- [ ] The clean code principles are respected ([CleanCode](https://clean-code-developer.com/virtues/))
- [ ] All new code/logic is implemented on the right spot / "Should this be done here?"
- [ ] UI changes fits into the layout
- [ ] The UI views / components are responsive (mobile views)
- [ ] Correct translations are available

### Checklist for branch merge request (not required for forks)

- [ ] Branch is green/blue on [Jenkins](https://oss.cloudogu.com/jenkins/)
- [ ] Quality Gate passed on [SonarQube](https://sonarcloud.io/organizations/scm-manager/projects)

### CODE REVIEW

**Comment by pfeuffer** [comment]
For me this looks like a workaround. It looks like the problem is, that we do not handle the configured default branch well. We shouldn't have the issue that the default branch is empty in the configuration.

**Comment by eheimbuch** [comment]
> For me this looks like a workaround. It looks like the problem is, that we do not handle the configured default branch well. We shouldn't have the issue that the default branch is empty in the configuration. Further more the default branch configured globally is ignored.

But what if an repository has been pushed, that doesn't even contain the globally configured default branch. Imho we should show the actual default branch of the repository and that is what this fix is about.

**Comment by pfeuffer** [comment]
> > For me this looks like a workaround. It looks like the problem is, that we do not handle the configured default branch well. We shouldn't have the issue that the default branch is empty in the configuration. Further more the default branch configured globally is ignored.
> 
> But what if an repository has been pushed, that doesn't even contain the globally configured default branch. Imho we should show the actual default branch of the repository and that is what this fix is about.

Sorry, forgot about the constraints from the global configuration.

File: scm-plugins/scm-git-plugin/src/main/js/RepositoryConfig.tsx [hunk 1]
```diff
@@ -101,13 +101,22 @@ class RepositoryConfig extends React.Component<Props, State> {
...
+            changed: false
+          },
+          () => {
+            if (!this.state.selectedBranchName) {
+              this.setState({
+                selectedBranchName: this.state.branches?.filter((b: Branch) => b.defaultBranch)[0]?.name
+              });
+            }
+          }
+        )
```
Comment (pfeuffer) [inline scm-plugins/scm-git-plugin/src/main/js/RepositoryConfig.tsx:119]: Wouldn't it be easier to make the check whether we have a default branch from the payload before we change the state? Maybe something like this:
```suggestion
      .then(payload => {
        const defaultBranch = payload.defaultBranch
          ? payload.defaultBranch
          : this.state.branches?.filter((b: Branch) => b.defaultBranch)[0]?.name;
        this.setState({
          selectedBranchName: defaultBranch,
          nonFastForwardDisallowed: payload.nonFastForwardDisallowed,
          disabled: !payload._links.update,
          loadingDefaultBranch: false,
          changed: false
        });
      })
```

**Review by pfeuffer — COMMENTED** [review]
Small simplification possible

**Pull request merged**